internal/elasticsearch/enrich/resource.go

```diff
@@ -0,0 +1,49 @@
+package enrich
+
+import (
+	"context"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+)
+
+func NewEnrichPolicyResource() resource.Resource {
+	return &enrichPolicyResource{}
+}
+
+func NewEnrichPolicyDataSource() datasource.DataSource {
```

Comment:
It would be easier to read this code if the data source functionality in this file was moved to its own `data_source.go` file.